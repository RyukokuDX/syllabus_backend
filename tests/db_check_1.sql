-- 数理の科目で科目名が「情報基礎」の科目名、シラバスコード、シラバスID、開講曜日、時限を表示
-- File Version: v2.0.0
-- Project Version: v2.1.6
-- Last Updated: 2025-07-02

-- 数理の科目で科目名が「情報基礎」の基本情報と講義時間を取得
SELECT DISTINCT
    sn.name AS 科目名,
    sm.syllabus_code AS シラバスコード,
    sm.syllabus_id AS シラバスID,
    lt.day_of_week AS 開講曜日,
    lt.period AS 時限
FROM syllabus_master sm
JOIN syllabus s ON sm.syllabus_id = s.syllabus_id
JOIN subject_name sn ON s.subject_name_id = sn.subject_name_id
JOIN lecture_time lt ON sm.syllabus_id = lt.syllabus_id
JOIN syllabus_faculty sf ON sm.syllabus_id = sf.syllabus_id
JOIN faculty f ON sf.faculty_id = f.faculty_id
WHERE f.faculty_name LIKE '%数理%'
  AND sn.name LIKE '%情報基礎%'
ORDER BY sn.name, sm.syllabus_code, lt.day_of_week, lt.period;

-- JSONBキャッシュから数理の科目で科目名が「情報基礎」のデータを検索
SELECT 
    sc.subject_name_id,
    sc.cache_data->>'科目名' AS 科目名,
    sc.cache_data->'開講情報' AS 開講情報
FROM syllabus_cache sc
WHERE sc.cache_name = 'subject_syllabus_cache'
  AND sc.cache_data->>'科目名' LIKE '%情報基礎%'
  AND EXISTS (
    SELECT 1
    FROM jsonb_array_elements(sc.cache_data->'開講情報') AS opening_info
    CROSS JOIN jsonb_array_elements(opening_info->'シラバス') AS syllabus
    CROSS JOIN jsonb_array_elements_text(syllabus->'対象学部課程') AS faculty
    WHERE faculty LIKE '%数理%'
  )
ORDER BY sc.cache_data->>'科目名'; 