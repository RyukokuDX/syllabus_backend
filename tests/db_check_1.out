SQLファイルを実行中: tests/db_check_1.sql
      科目名       | シラバスコード | シラバスid | 開講曜日 | 時限 
-------------------+----------------+------------+----------+------
 情報基礎          | Y000001010     |          1 | 木       |    4
 情報基礎          | Y000001010     |          1 | 木       |    5
 数理情報基礎演習A | Y001002010     |         32 | 金       |    3
 数理情報基礎演習B | Y001005010     |         35 | 金       |    3
(4 行)

 subject_name_id |      科目名       |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      開講情報                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
-----------------+-------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
             291 | 情報基礎          | [{"年": 2025, "シラバス": [{"単位": 2, "学期": "1Q", "成績": [{"備考": "", "割合": 100, "項目": "平常点", "評価方法": "各回の授業で出される課題の達成度で評価します。"}], "担当": [{"役割": "担当", "教員名": "中野 浩"}], "時限": [4, 5], "曜日": "木", "参考書": [], "教科書": [], "対象学部課程": ["先端理工学部", "数理·情報科学課程"], "成績コメント": "", "参考書コメント": "特になし", "教科書コメント": "特になし"}, {"単位": 2, "学期": "1Q", "成績": [{"備考": "", "割合": 50, "項目": "レポート", "評価方法": "演習課題に対するレポート"}, {"備考": "", "割合": 50, "項目": "小テスト", "評価方法": "最終週にテストを実施する"}], "担当": [{"役割": "担当", "教員名": "植田 祥明"}, {"役割": "担当", "教員名": "山本 哲男"}], "時限": [4, 5], "曜日": "火", "参考書": [], "教科書": [], "対象学部課程": ["先端理工学部", "知能情報メディア課程"], "成績コメント": "", "参考書コメント": "特になし", "教科書コメント": "特になし"}, {"単位": 2, "学期": "1Q", "成績": [{"備考": "", "割合": 60, "項目": "レポート", "評価方法": "授業時間内に課題に取り組む"}, {"備考": "", "割合": 40, "項目": "平常点", "評価方法": ""}], "担当": [{"役割": "担当", "教員名": "小堀 聡"}], "時限": [3, 4], "曜日": "木", "参考書": [], "教科書": [{"ISBN": null, "値段": null, "書名": "龍谷大学情報システムガイドブック", "著者": "龍谷大学情報メディアセンター編"}], "対象学部課程": ["先端理工学部", "電子情報通信課程"], "成績コメント": "平常点40点、レポート60点、合計100点満点で評価する。\n　オンライン授業の場合も変更はしない。", "参考書コメント": "その他の参考書については、授業の中で随時紹介する。", "教科書コメント": "『龍谷大学情報システムガイドブック』については無償配布されるPDFを使用する予定である。"}, {"単位": 2, "学期": "1Q", "成績": [{"備考": "", "割合": 60, "項目": "レポート", "評価方法": "講義に関する演習課題の提出"}, {"備考": "", "割合": 40, "項目": "小テスト", "評価方法": "第8回目の講義にて講義全般のまとめのテスト（実技）を行う。"}], "担当": [{"役割": "担当", "教員名": "野口 佳樹"}], "時限": [4, 5], "曜日": "月", "参考書": [], "教科書": [], "対象学部課程": ["先端理工学部", "機械工学·ロボティクス課程"], "成績コメント": "原則として全出席、レポート全提出を合格の条件とする。レポート点、小テスト（まとめのテスト）にて評価する。", "参考書コメント": "特になし", "教科書コメント": "特になし"}, {"単位": 2, "学期": "1Q", "成績": [{"備考": "", "割合": 100, "項目": "レポート", "評価方法": "毎回提示する課題"}], "担当": [{"役割": "担当", "教員名": "青井 芳史"}, {"役割": "担当", "教員名": "岩澤 哲郎"}, {"役割": "担当", "教員名": "河内 岳大"}], "時限": [4, 5], "曜日": "木", "参考書": [], "教科書": [], "対象学部課程": ["先端理工学部", "応用化学課程"], "成績コメント": "", "参考書コメント": "適宜指示する。", "教科書コメント": "適宜指示する。"}, {"単位": 2, "学期": "1Q", "成績": [{"備考": "", "割合": 50, "項目": "レポート", "評価方法": "講義中に指示する提出物の内容で評価する"}, {"備考": "", "割合": 50, "項目": "平常点", "評価方法": "毎回の出席を前提に、講義中の指示をこなすことで評価する"}], "担当": [{"役割": "担当", "教員名": "丸山 敦"}], "時限": [1, 2], "曜日": "木", "参考書": [], "教科書": [], "対象学部課程": ["先端理工学部", "環境生態工学課程", "環境科学課程"], "成績コメント": "", "参考書コメント": "特になし", "教科書コメント": "特になし"}]}]
             321 | 数理情報基礎演習A | [{"年": 2025, "シラバス": [{"単位": 2, "学期": "前期", "成績": [{"備考": "", "割合": 100, "項目": "平常点", "評価方法": "科目に対する取り組みの姿勢を評価します。"}], "担当": [{"役割": "担当", "教員名": "飯田 晋司"}, {"役割": "担当", "教員名": "馬 青"}, {"役割": "担当", "教員名": "村川 秀樹"}], "時限": [3], "曜日": "金", "参考書": [], "教科書": [], "対象学部課程": ["先端理工学部", "数理·情報科学課程"], "成績コメント": "", "参考書コメント": "微積分及び演習Ⅰ、線形代数及び演習Ⅰ、情報基礎、プログラミング及び実習Ⅰの配布資料やテキスト", "教科書コメント": "特になし"}]}]
             322 | 数理情報基礎演習B | [{"年": 2025, "シラバス": [{"単位": 2, "学期": "後期", "成績": [{"備考": "", "割合": 100, "項目": "平常点", "評価方法": "科目に対する取り組みの姿勢を評価します。"}], "担当": [{"役割": "担当", "教員名": "樋口 三郎"}, {"役割": "担当", "教員名": "藤原 和将"}, {"役割": "担当", "教員名": "村川 秀樹"}], "時限": [3], "曜日": "金", "参考書": [], "教科書": [], "対象学部課程": ["先端理工学部", "数理·情報科学課程"], "成績コメント": "", "参考書コメント": "微積分及び演習Ⅱ、線形代数及び演習Ⅱ、データ分析、プログラミング及び実習Ⅱ、情報処理システムⅠの配布資料やテキスト", "教科書コメント": "特になし"}]}]
(3 行)

